crates/forge/bin/cmd/flatten.rs

```diff
@@ -38,10 +39,27 @@ impl FlattenArgs {
 
         let config = build_args.try_load_config_emit_warnings()?;
 
-        let paths = config.project_paths();
         let target_path = dunce::canonicalize(target_path)?;
-        let flattened =
-            paths.flatten(&target_path).map_err(|err| eyre::eyre!("Failed to flatten: {err}"))?;
+
+        // We need to provide Flattener with compiled output of target and all of its imports.
+        let project = config.ephemeral_no_artifacts_project()?;
+        let sources = Source::read_all(vec![target_path.clone()])?;
+        let (sources, _) = Graph::resolve_sources(&project.paths, sources)?.into_sources();
+
+        let compiler_output = ProjectCompiler::new().files(sources.into_keys()).compile(&project);
```

Comment:
can't we just use `compile` without doing the Source and Graph stuff?

Reply:
yeah, it seems that we can. I belive that was a workaround I used before #6935 which is no more needed